monitor/core.go

```diff
@@ -190,7 +206,7 @@ func (qm *QueueMonitor) consumeMessage(ec chan error, pConsumer sarama.Partition
 	defer func() {
```

Comment:
May be this hack could also be avoided with the help of context? Ex, once close message is received, skip all writes. And only when all closes are done, close the channel. I am just explaining the thought. Not sure about details.

Reply:
Yeah, will try to implement this in a cleaner way. I'll raise another PR for this.